### DISCUSSION TITLE
Async vs Sync handlers

### DISCUSSION THREAD

**Discussion by valkrypton** [rwf2/Rocket#2884] [Questions] [answered]
I am implementing api endpoints for my web app, and I am not sure whether to make the handlers async or sync. The endpoints mostly consist of calls to other function that do basic CRUD operations. I have made them async right now purely based on intuition, but I don't actually know what should I opt for as having both async and sync handlers seems to not make a difference.

**Comment by jbcurtin** [comment]
Hi @valkrypton ,

 If you use `tokio` to manage the runtime. You probably will never notice a difference in performance between sync/async handlers until there is a lot of file or network io.

**Comment by the10thWiz** [comment, marked as answer]
Rocket starts a `tokio` runtime, and every method is executed on the async runtime. The general advice I would give, is that you should always use `tokio` types for doing any kind of IO, especially network/file IO. Handlers only need to be marked as async if you use any of these types, and call `.await` inside them. Clippy has a warning for `async` functions that don't use `.await`, but it's up to you whether you remove the `async` marker from your functions that don't need it.
